backend/LexBoxApi/Services/HgService.cs

```diff
@@ -38,6 +39,7 @@ public HgService(IOptions<HgConfig> options, IHttpClientFactory clientFactory, I
 
     public static string PrefixRepoRequestPath(string code) => $"{code[0]}/{code}";
     private string PrefixRepoFilePath(string code) => Path.Combine(_options.Value.RepoPath, code[0].ToString(), code);
+    private string GetTempRepoPath(string code, string reason) => Path.Combine(_options.Value.RepoPath, TEMP_REPO_FOLDER, $"{code}__{reason}__{FileUtils.ToTimestamp(DateTimeOffset.UtcNow)}");
```

Comment:
My one concern on first read is that I could see wanting to call GetTempRepoPath twice with the same timestamp, so it might be good to allow passing in a timestamp rather than always getting UtcNow. However, that's an overload that can be created when it's needed, so there's no need to add it now. Probably YAGNI.

Reply:
yeah we don't need it so lets keep it simple